### PR TITLE
[App Config] `az appconfig kv export/import/restore`: Update key-value diffing and preview

### PR DESCRIPTION
**Pull request by albertofori**
**Related command**

`az appconfig kv import`
`az appconfig kv export`
`az appconfig kv restore`

**Description**

This change updates and unifies how the diff for bulk operations is determined and displayed based on the source/destination. This unifies diffing logic and differentiates preview serialization based on the source/destination.

**Testing Guide**

**History Notes**

{App Config}: `az appconfig kv export/import/restore`: Update key-value diffing and preview

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Config

**Comment by albertofori** [comment]
@avanigupta could you please take a look at this? Thanks!

**Comment by zhoxing-ms** [comment]
@albertofori Could you please resolve these conflicts?

**Comment by zhoxing-ms** [comment]
@albertofori Please note that we are launching the release for this sprint this week. Please resolve all comments by tomorrow, otherwise the release of this PR will have to be postponed to the next sprint

**Comment by zhoxing-ms** [comment]
@albertofori Any update for this PR?

**Comment by zhoxing-ms** [comment]
@albertofori Please note that we are launching the release for this sprint this week. Please resolve all comments by tomorrow, otherwise the release of this PR will have to be postponed to the next sprint (on 08-01)

**Comment by albertofori** [comment]
> @albertofori Please note that we are launching the release for this sprint this week. Please resolve all comments by tomorrow, otherwise the release of this PR will have to be postponed to the next sprint (on 08-01)

@

> @albertofori Any update for this PR?

@zhoxing-ms , please we would want to postpone this PR to the next sprint. Thanks!

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py [hunk 1]
```diff
@@ -311,17 +320,29 @@ def export_config(cmd,
...
-    need_kv_change = __print_kv_preview(
-        old_json=__serialize_kv_list_to_comparable_json_object(keyvalues=dest_kvs, level=destination),
-        new_json=__serialize_kv_list_to_comparable_json_object(keyvalues=src_kvs, level=destination),
-        yes=yes)
+    kv_comparer = KVComparer(
+        src_kvs=src_kvs,
+        compare_fields=CompareFieldsMap[destination],
+        preserve_labels=destination == "appconfig" and preserve_labels,
+        label=dest_label)
+
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:328]: do we need to pass content type here?

File: src/azure-cli/azure/cli/command_modules/appconfig/_diff_utils.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+        return lookup
+
+    @staticmethod
+    def _try_parse_json_value(value_string):
+        try:
+            return json.loads(value_string)
+
+        except Exception:  # pylint: disable=broad-except
+            return value_string
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/_diff_utils.py:82]: Catch specific exceptions that `json.loads` might throw

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py (hunk 1 above)
Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:328]: The content type is not needed here as we do not specify destination content type for key-value export.

File: src/azure-cli/azure/cli/command_modules/appconfig/_diff_utils.py (hunk 2 above)
Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/_diff_utils.py:82]: This has been updated, thanks!

**Review by avanigupta — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix conflicting files.

**Comment by albertofori** [comment]
> Please fix conflicting files.

@yonzhan Please the conflicts have been fixed. Thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**